akka-actor/src/main/scala/akka/io/dns/internal/AsyncDnsResolver.scala

```diff
@@ -136,90 +146,39 @@ private[io] final class AsyncDnsResolver(
                   log.info("Resolve of {} failed. Trying next name server {}", name, t.getMessage)
               }
               resolveWithResolvers(name, requestType, tail)
-          }
+          }(ExecutionContexts.parasitic)
       }
     }
 
-  private def sendQuestion(resolver: ActorRef, message: DnsQuestion): Future[Answer] = {
-    val result = (resolver ? message).mapTo[Answer]
-    result.failed.foreach { _ =>
-      resolver ! DropRequest(message.id)
-    }
-    result
-  }
-
-  private def resolveWithSearch(
+  private def resolvePromise(
```

Comment:
Do I understand it correctly that the only reason for going back to the actor via `Internal.Resolve` is that the `nextId()` is needed here, or is there something else that requires this to be running in the actor?

If that is the case, would it be better to change the requestId counter to an a thread safe counter (AtomicInteger) and stay in the Future land in the companion object methods?